### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by peterdgreat**
## Testing Components and Logic

- I set up React Testing Library
- Wrote unit tests for the files operate.js and calculate.js using Jest
- Created unit tests for all React components

### CODE REVIEW

File: src/test/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import calculate, { isNumber } from '../logic/calculate';
+
+// check for the function isNumber
+it('isNumber', () => {
+  expect(typeof isNumber).toBe('function');
+});
+
+it('calculate', () => {
+  expect(typeof calculate).toBe('function');
+});
```
Comment (helman101) [inline src/test/calculate.test.js]: - [ ] Please test all the functionalities and scenarios, here you only are making sure that these two are functions.
   - But what should they return on each situation, how should they work?
   - Kindly add more meaningful tests 😄

File: src/test/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  expect(operate(5, 3, '-')).toBe('2');
+});
+
+it('should multiply two numbers', () => {
+  expect(operate(2, 3, 'x')).toBe('6');
+});
+
+it('should divide two numbers', () => {
+  expect(operate(6, 3, '÷')).toBe('2');
+});
```
Comment (helman101) [inline src/test/operate.test.js:17]: - [ ] You forgot to test the `%` operation, kindly add a test for this. 🚀

File: src/test/page.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+describe('Testing Container page with jest snapshot', () => {
+  it('renders correctly', () => {
+    const tree = renderer
+      .create(<App />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (helman101) [inline src/test/page.test.js]: - [ ] ⭐ **All** ⭐  your components should have snapshots, not only the `App` component, kindly add the remaining snapshots

File: src/test/interaction.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Calculator from '../components/Calculator';
+
+describe('Calculator', () => {
+  it('should render the calculator', () => {
+    render(<Calculator />);
+    const button = screen.getByRole('button', { name: '1' });
+    expect(button).toBeInTheDocument();
+  });
```
Comment (helman101) [inline src/test/interaction.test.js]: - [ ] This test is not testing user interaction, please replace this and add more interaction test 😄

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

Hi @peterdgreat 👋, 

Nice work so far on this project 😄, but there are still some changes to do.

### Required Changes ♻️

- [ ] _Check the comments below_

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rukundoeric — APPROVED** [review]
Hi @peterdgreat,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**